azure/Kqlmagic/kusto_client.py

```diff
@@ -80,19 +94,47 @@ def __init__(self, conn_kv:dict):
         authority : 'microsoft.com', optional
             In case your tenant is not microsoft please use this param.
         """
+        cloud = options.get("cloud")
 
         cluster_name = conn_kv[ConnStrKeys.CLUSTER]
-        data_source = cluster_name if cluster_name.find("://") >= 0 else self._DATA_SOURCE_TEMPLATE.format(cluster_name)
+
+        if cluster_name.find("://") >= 0:
+            data_source = cluster_name
+            cloud = cloud or self.getCloudFromHTTP(data_source)
```

Comment:
Here cloud will always have value and can't be none, so there is no point to try to retreive it from the data_source. So just remove this line. We won't try to derive it from his data_source url, he will have to set the cloud option properly.